File: Reqnroll/Chapter18/18.2-ResetBeforePattern/Before/WIMP.Specs/Drivers/StubDatabaseDriver.cs
namespace WIMP.Specs.Drivers;

public class StubDatabaseDriver : IDatabaseDriver
{
    public Task CreateDatabase()
    {
        //nop
        return Task.CompletedTask;
    }

    public Task DropDatabase()
    {
        //nop
        return Task.CompletedTask;
    }
}

File: CucumberJVM/Chapter21/21.1-TestEnvironmentInitializationPattern/Before/src/test/java/com/wimp/app/specs/drivers/StubDatabaseDriver.java
package com.wimp.app.specs.drivers;

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.stereotype.Component;

@Component
@ConditionalOnProperty(name = "test.database.use-stub", havingValue = "true")
public class StubDatabaseDriver implements DatabaseDriver {

    @Override
    public void emptyDatabase() {
        // nop
    }
}
